server/constants.py

```diff
@@ -10,6 +10,9 @@
               'revision']
 API_PREFIX = '/api'
 
+COURSE_ENDPOINT_FORMAT = '^\w+/\w+/\w+$'
+ASSIGNMENT_ENDPOINT_FORMAT = COURSE_ENDPOINT_FORMAT[:-1] + '/\w+$'
```

Comment:
The one in converters.py also takes care of forbidden assignment names, etc, so is_valid_endpoint would just be a reflex match.